### PR TITLE
Fix issue with slashes being turned into backslashes on Windows

### LINKED ISSUE #12745: Slashes in parametrization values get turned into backslashes on Windows
With pytest 8.3.2 as well as the current `main`, when creating a `repro/test_x.py` in the pytest repo:

```python
import pytest

@pytest.mark.parametrize("a", ["x/y", "/", "//"])
def test_x(a):
    assert False
```

and running `pytest test_x.py` inside `repro/`, both in the `-v` output and short test summary, the tests show up as:

```
test_x.py::test_x[x\y]
test_x.py::test_x[\]
test_x.py::test_x[\]
```

The IDs show up correctly in `--collect-only`, and they show correctly when `test_x.py` is not in a subfolder, or when pytest is not invoked in the subfolder.

### PR DESCRIPTION
**Pull request by nauman897**
Fix slash handling in nodeid paths for Windows environments.

- Resolved issue with backslashes being incorrectly converted in `nodeid` paths on Windows.
- This change closes #12745.

### CODE REVIEW

**Comment by nauman897** [comment]
I’ve tackled the issue with backslashes getting messed up in `nodeid` paths on Windows by isolating the path and `nodeid` parts, handling them separately, and then putting them back together. This should take care of the problem, including a few tricky edge cases with mixed slashes and special characters.

I haven’t added unit tests yet—figured I’d wait to see if the approach looks good to you all. If it does, I’ll go ahead and write the tests to make sure everything’s covered.

File: src/_pytest/config/__init__.py [hunk 1]
```diff
@@ -1178,8 +1178,13 @@ def notify_exception(
...
         if self.invocation_params.dir != self.rootpath:
-            fullpath = self.rootpath / nodeid
-            nodeid = bestrelpath(self.invocation_params.dir, fullpath)
+            base_path_part = nodeid.split("::")[0]
+            nodeid_part = "::".join(nodeid.split("::")[1:])
+
+            fullpath = self.rootpath / base_path_part
+            relative_path = bestrelpath(self.invocation_params.dir, fullpath)
+
+            nodeid = f"{relative_path}::{nodeid_part}"
```
Comment (dongfangtianyu) [inline src/_pytest/config/__init__.py:1187]: Hi, @nauman897 This PR looks good !
A small suggestion about the coding stylem, which can avoid repeated use of the split method and accessing values ​​through indexes (of course, this is trivial)
```suggestion
            base_path_part, *nodeid_part = nodeid.split("::")
            # Only process path part
            fullpath = self.rootpath / base_path_part
            relative_path = bestrelpath(self.invocation_params.dir, fullpath)

            nodeid = "::".join([relative_path, *nodeid_part])
```

Comment (nauman897) [reply src/_pytest/config/__init__.py]: Hey @dongfangtianyu ,
Thanks for the suggestion. I agree this looks better

**Review by dongfangtianyu — APPROVED** [review]
Thank you @nauman897  for this PR.
I have debugged and verified it locally, and everything is fine.

File: testing/test_terminal.py [hunk 2]
```diff
@@ -3067,3 +3067,28 @@ def test_pass():
             "*= 1 xpassed in * =*",
         ]
     )
+
+
+class TestNodeIDHandling:
+    def test_nodeid_handling_windows_paths(self, pytester: Pytester) -> None:
```
Comment (nicoddemus) [inline testing/test_terminal.py:3073]: I just tested this on Windows, on the `main` branch, and it passes... does this fail only on Linux?

Comment (nauman897) [reply testing/test_terminal.py:3073]: Hey @nicoddemus ,
The problem specifically occurs under these circumstances as mentioned in the issue:
`The IDs show up correctly in --collect-only, and they show correctly when test_x.py is not in a subfolder, or when pytest is not invoked in the subfolder.`

Comment (The-Compiler) [reply testing/test_terminal.py:3073]: But then the test case should make sure to replicate that scenario, no?

Comment (nauman897) [reply testing/test_terminal.py:3073]: Yeah. I'll fix that

Comment (nauman897) [reply testing/test_terminal.py:3073]: I believe @dongfangtianyu has already taken care of it: https://github.com/nauman897/pytest/pull/1

Comment (nauman897) [reply testing/test_terminal.py]: Thanks @dongfangtianyu for the help with these tests.
I can confirm that:
1. The tests fail with the previous logic in `cwd_relative_nodeid`:
```
=========================== short test summary info ===========================
FAILED test_foo.py::test_x[x\y] - assert False
FAILED test_foo.py::test_x[C:\path] - assert False
FAILED test_foo.py::test_x[\] - assert False
FAILED test_foo.py::test_x[C:\path] - assert False
FAILED test_foo.py::test_x[a::b\] - assert False
============================== 5 failed in 0.11s ==============================
```
2. The tests pass after the patch is applied.

I had to make a minor adjustment to the new tests, as Python 3.11 and later requires passing the `encoding_type` param while calling the `write_text` method.

Comment (dongfangtianyu) [reply testing/test_terminal.py]: That's cool @nauman897  !


> 2. The tests pass after the patch is applied.
> 
> ```
> test_terminal.py::TestNodeIDHandling::test_nodeid_handling_windows_paths PASSED [100%]
> ```

This is a more intuitive output:
```
test_foo.py::test_x[x/y] FAILED                                          [ 20%]
test_foo.py::test_x[C:/path] FAILED                                      [ 40%]
test_foo.py::test_x[\\] FAILED                                           [ 60%]
test_foo.py::test_x[C:\\path] FAILED                                     [ 80%]
test_foo.py::test_x[a::b/] FAILED                                        [100%]
```

**Review by nicoddemus — APPROVED** [review]
Thanks folks, we greatly appreciate it!

**Pull request merged**